ios/FluentUI/Core/Theme/Tokens/ControlTokenSet.swift

```diff
@@ -0,0 +1,147 @@
+    ///
+    /// - Returns: The default value for a given token value.
+    func defaultValue(_ token: T) -> ControlTokenValue {
+        preconditionFailure("Override defaultValue in your ControlTokenSet")
+    }
+
+    /// Prepares this token set by installing the current `FluentTheme` and then allowing the control to perform
+    /// additional configuration (e.g. setting `style` or `size` properties) by invoking the `configuration` callback.
+    ///
+    /// - Parameter fluentTheme: The current `FluentTheme` for the control's environment.
+    /// - Parameter configuration: An optional callback for the control to perform additional configuration.
+    func update(_ fluentTheme: FluentTheme,
+                _ configuration: (() -> Void)? = nil) {
+        if fluentTheme != self.fluentTheme {
+            self.fluentTheme = fluentTheme
+        }
+
+        if let configuration = configuration {
+            configuration()
+        }
+    }
+
+    @Published var fluentTheme: FluentTheme = FluentTheme.shared
+
+    @Published private var valueOverrides: [T: ControlTokenValue]?
+}
+
+/// Union-type enumeration of all possible token values to be stored by a `ControlTokenSet`.
+public enum ControlTokenValue {
+    case float(() -> CGFloat)
+    case dynamicColor(() -> DynamicColor)
+    case fontInfo(() -> FontInfo)
+    case shadowInfo(() -> ShadowInfo)
+
+    var float: CGFloat {
+        if case .float(let float) = self {
+            return float()
+        } else {
+            assertionFailure("Cannot convert token to CGFloat: \(self)")
+            return 0.0
+        }
+    }
+
+    var dynamicColor: DynamicColor {
+        if case .dynamicColor(let dynamicColor) = self {
+            return dynamicColor()
+        } else {
+            assertionFailure("Cannot convert token to DynamicColor: \(self)")
+            return DynamicColor(light: ColorValue(0xE3008C))
+        }
+    }
+
+    var fontInfo: FontInfo {
+        if case .fontInfo(let fontInfo) = self {
+            return fontInfo()
+        } else {
+            assertionFailure("Cannot convert token to FontInfo: \(self)")
+            return FontInfo(size: 0.0)
+        }
+    }
+
+    var shadowInfo: ShadowInfo {
+        if case .shadowInfo(let shadowInfo) = self {
+            return shadowInfo()
+        } else {
+            assertionFailure("Cannot convert token to ShadowInfo: \(self)")
+            let defaultColor = DynamicColor(light: ColorValue(0xE3008C))
+            return ShadowInfo(colorOne: defaultColor,
+                              blurOne: 10.0,
+                              xOne: 10.0,
+                              yOne: 10.0,
+                              colorTwo: defaultColor,
+                              blurTwo: 10.0,
+                              xTwo: 10.0,
+                              yTwo: 10.0)
+        }
+    }
+}
+
+#if DEBUG
+extension ControlTokenValue: CustomStringConvertible {
```

Comment:
maybe a todo is a unit test to figure out if the resolver actually works correctly?